sormas-app/app/src/main/java/de/symeda/sormas/app/caze/edit/CaseValidator.java

```diff
@@ -42,224 +41,192 @@
-			false);
-	}
+    static void initializeGermanCaseClassificationValidation(
+            Case caze,
+            CaseClassification caseClassification,
+            FragmentCaseEditLayoutBinding contentBinding) {
+        if (ConfigProvider.isConfiguredServer(CountryHelper.COUNTRY_CODE_GERMANY)) {
+            final CaseDtoHelper caseDtoHelper = new CaseDtoHelper();
+            final CaseDataDto caseDataDto = caseDtoHelper.adoToDto(caze);
+            final boolean hasPositiveTestResult = DatabaseHelper.getSampleDao().hasPositiveTestResult(caze);
+            final boolean validCaseClassification =
+                    GermanCaseClassificationValidator.isValidGermanCaseClassification(caseClassification, caseDataDto, hasPositiveTestResult);
+
+            if (validCaseClassification) {
+                contentBinding.caseDataCaseClassification.disableErrorState();
+            } else {
+                contentBinding.caseDataCaseClassification.enableErrorState(R.string.validation_case_classification);
+            }
+
+            contentBinding.caseDataCaseClassification.setValidationCallback(() -> {
+                if (validCaseClassification) {
+                    return false;
+                }
+                return true;
+            });
+        }
+    }
+
+    static void initializePortHealthInfoValidation(final FragmentCaseEditPortHealthInfoLayoutBinding contentBinding, final Case caze) {
+        if (contentBinding.portHealthInfoDepartureDateTime.getVisibility() == View.GONE
+                || contentBinding.portHealthInfoArrivalDateTime.getVisibility() == View.GONE) {
+            return;
+        }
+
+        ResultCallback<Boolean> departureCallback = () -> {
+            if (contentBinding.portHealthInfoDepartureDateTime.getValue() != null
+                    && contentBinding.portHealthInfoArrivalDateTime.getValue() != null) {
+                if (DateHelper.isDateAfter(contentBinding.portHealthInfoDepartureDateTime.getValue(), contentBinding.portHealthInfoArrivalDateTime.getValue())) {
+                    contentBinding.portHealthInfoDepartureDateTime.enableErrorState(
+                            I18nProperties.getValidationError(
+                                    Validations.beforeDate,
+                                    contentBinding.portHealthInfoDepartureDateTime.getCaption(),
+                                    contentBinding.portHealthInfoArrivalDateTime.getCaption()));
+                    return true;
+                }
+            }
+
+            return false;
+        };
+
+        ResultCallback<Boolean> arrivalCallback = () -> {
+            if (contentBinding.portHealthInfoArrivalDateTime.getValue() != null
+                    && contentBinding.portHealthInfoDepartureDateTime.getValue() != null) {
+                if (DateHelper.isDateAfter(contentBinding.portHealthInfoArrivalDateTime.getValue(), contentBinding.portHealthInfoDepartureDateTime.getValue())) {
```

Comment:
Should be `isDateBefore` because arrival date needs to be after departure date

Reply:
Yes, that's why it enables an error state when the arrival is after the departure